app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/firestore/CachedFirestoreDatabaseManagement.kt

```diff
@@ -0,0 +1,56 @@
+package com.github.HumanLearning2021.HumanLearningApp.firestore
+
+import com.github.HumanLearning2021.HumanLearningApp.model.CategorizedPicture
+import com.github.HumanLearning2021.HumanLearningApp.model.Category
+import com.github.HumanLearning2021.HumanLearningApp.model.Converters
+import com.github.HumanLearning2021.HumanLearningApp.offline.OfflineCategorizedPicture
+import com.github.HumanLearning2021.HumanLearningApp.offline.PictureRepository
+
+class CachedFirestoreDatabaseManagement internal constructor(
+    databaseService: FirestoreDatabaseService
+): FirestoreDatabaseManagement(databaseService) {
```

Comment:
Kotlin has [language-level support for delegation](https://kotlinlang.org/docs/delegation.html) and delegation seems to make a lot more sense in this situation. Could you investigate that?

Reply:
I will look into it, thanks